### PR TITLE
[IMP] web, account: improve carets visibility

### PR DESCRIPTION
**Pull request by BastienFafchamps**
This PR improves various carets of the autocomplete dropdown and the list view sortable columns. It improves it's visibility and saves space when it overlaps with text.

Task: [4170216](https://www.odoo.com/odoo/project/133/tasks/4170216)

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yosuanicolaus** [comment]
Hello :wave: is this ready for review? Do you a task?

**Comment by BastienFafchamps** [comment]
Hello, yes this is ready for review just now!

The change regarding account simply makes the button added in many2X smaller to save space and make it look a bit more consistent with other buttons.

**Review by yosuanicolaus — APPROVED** [review]
it's a bit out of my scope, so I'll let the JS team review it, or if you have a dedicated reviewer here :)

Accounting side of the code LGTM, 
Just one thing: don't forget to add your change summary and task-id in your commit message

File: addons/web/static/src/views/list/list_renderer.scss [hunk 1]
```diff
@@ -89,12 +90,46 @@
...
+                    right: .2rem;
+                    padding: 0 .2rem;
+                }
+
+                &.table-active .o_list_sortable_caret {
+                    background: var(--ListRenderer-thead-bg-active-color);
+                }
+
+                .o_list_header_label_spacer {
+                    width: 0rem;
```
Comment (pparidans) [inline addons/web/static/src/views/list/list_renderer.scss:119]: Not necessary to mention the unit when value is `0` :wink:

File: addons/web/static/src/views/list/list_renderer.xml [hunk 2]
```diff
@@ -28,12 +28,13 @@
...
                                 <t t-set="isNumeric" t-value="isNumericColumn(column)"/>
-                                    <div t-att-class="{'d-flex': true, 'flex-row-reverse': shouldReverseHeader(column)}"
+                                    <div class="d-flex"
                                         t-att-data-tooltip-template="isDebugMode ? 'web.FieldTooltip' : 'web.ListHeaderTooltip'"
                                         t-att-data-tooltip-info="makeTooltip(column)">
-                                        <span class="d-block min-w-0 text-truncate flex-grow-1" t-att-class="isNumeric ? 'o_list_number_th' : ''"
+                                        <span class="d-block min-w-0 text-truncate flex-grow-1 flex-shrink-1" t-att-class="isNumeric ? 'o_list_number_th' : ''"
                                               t-esc="column.label"/>
-                                        <i t-att-class="getSortableIconClass(column)"/>
+                                        <span class="d-block o_list_header_label_spacer"></span>
```
Comment (pparidans) [inline addons/web/static/src/views/list/list_renderer.xml:36]: - `<span>` are non-semantic *inline* elements.
- `<div>` are non-semantic *block* elements.

So, `<span class="d-block">` is a longer way for `<div>` :wink: 
I think you can simplify (I haven't checked if it breaks some selector, though...)

File: addons/web/static/src/views/list/list_renderer.scss [hunk 3]
```diff
@@ -89,12 +90,46 @@
...
+                }
+
+                &.table-active .o_list_header_label_spacer {
+                    width: 1.2rem;
+                    max-width: 1.2rem;
+                    min-width: .4rem;
+
+                    flex-shrink: 0;
+                    flex-grow: 1;
+                    flex-basis: 0px;
```
Comment (pparidans) [inline addons/web/static/src/views/list/list_renderer.scss:130]: Same.

File: addons/web/static/src/views/list/list_renderer.scss [hunk 4]
```diff
@@ -3,6 +3,7 @@
     --ListRenderer-table-padding-x: #{$table-cell-padding-x-sm};
     --ListRenderer-thead-padding-y: #{map-get($spacers, 4)};
     --ListRenderer-thead-bg-color:  #{$o-webclient-background-color};
+    --ListRenderer-thead-bg-active-color:  #{$o-gray-200};
```
Comment (pparidans) [inline addons/web/static/src/views/list/list_renderer.scss:6]: double space

File: addons/web/static/src/views/form/form_controller.scss [hunk 5]
```diff
@@ -926,6 +926,7 @@
     // One2Many List views
     .o_field_widget .o_list_renderer {
         --ListRenderer-thead-bg-color:  #{$o-view-background-color};
+        --ListRenderer-thead-bg-active-color:  #{$o-form-lightsecondary};
```
Comment (pparidans) [inline addons/web/static/src/views/form/form_controller.scss:929]: double space

**Review by pparidans — CHANGES_REQUESTED** [review]

**Comment by BastienFafchamps** [comment]
Retargeted to 18.0 -> https://github.com/odoo/odoo/pull/181636

**Pull request closed without merging**